### PR TITLE
[#360] Add include support for Ruby DSL

### PR DESCRIPTION
**Pull request by llby**
This PR add include support for Ruby DSL.
And Fix bug parser include at the Windows.
Like this conditions
- First conf include absolute path
- Second(child) conf include relative path

Because `URI.parse` return `u.scheme = c`.
When Windows absolute path is `C:/projects/fluentd/test/tmp/config`.
Then used `open-uri` and basepath change `/`.

But after merge this PR.
There is a bug.
When It has some drives In the Windows.
Like a C: and D:
And upper case.
I think it is rare case.
If you need fix this bug.
Please make issue.

### CODE REVIEW

File: lib/fluent/config/dsl.rb [hunk 1]
```diff
@@ -96,6 +101,18 @@ def method_missing(name, *args, &block)
           self
         end
 
+        def include(*args)
+          ::Kernel.raise ::ArgumentError, "#{name} block requires arguments for include path" if args.nil? || args.size != 1
+          if args.first =~ /\.rb$/
+            path = File.expand_path(args.first)
+            data = File.read(path)
+            self.instance_eval(data, @proxy.include_basepath)
```
Comment (tagomoris) [inline lib/fluent/config/dsl.rb]: The 2nd argument of `instance_eval` should be the filename (`path`), not base path.

**Comment by tagomoris** [comment]
Almost looks good to me.

**Comment by nagata-logly** [comment]
@tagomoris 
Thank you your review
I got the `instance_eval` 2nd args

**Comment by nagata-logly** [comment]
@tagomoris dsl code is cool!!

**Comment by llby** [comment]
@tagomoris 
Thank you your review
I got the `instance_eval` 2nd args

and ... dsl code is cool!!

why test is failed??

**Comment by tagomoris** [comment]
Merged. Thank you for contribution!

**Comment by tagomoris** [comment]
This patch disables (overwrites) `Kernel.include`, but I think there are no users who wants to use mixin in configuration.

**Pull request merged**